### PR TITLE
Allow the user to restrict the rows to extract

### PR DESCRIPTION
**Pull request by hayssams**
## Summary
A new attribute ("where") has been added and is applied on extraction.

**PR Type:  Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/ai/starlake/extract/JDBCSchemas.scala [hunk 1]
```diff
@@ -96,7 +96,8 @@ case class JDBCSchema(
   fetchSize: Option[Int] = None,
   stringPartitionFunc: Option[String] = None,
   fullExport: Option[Boolean] = None,
-  sanitizeName: Option[Boolean] = None
+  sanitizeName: Option[Boolean] = None,
+  where: Option[String] = None
```
Comment (tiboun) [inline src/main/scala/ai/starlake/extract/JDBCSchemas.scala]: In load, I think we named this attribute `filter` and not `where` and they have the same feature.

File: src/main/scala/ai/starlake/extract/JdbcDbUtils.scala [hunk 2]
```diff
@@ -1278,11 +1278,12 @@ object JdbcDbUtils extends LazyLogging {
     auditColumns: Columns
   )(implicit settings: Settings): Try[Unit] = {
     val dataColumnsProjection = tableExtractDataConfig.columnsProjectionQuery(extractConfig.data)
+    val where = extractConfig.jdbcSchema.where.map(w => s"where $w").getOrElse("")
```
Comment (tiboun) [inline src/main/scala/ai/starlake/extract/JdbcDbUtils.scala]: We should rename it to extraCondition or rename above one to where in order to have the same name for both function

File: src/main/scala/ai/starlake/extract/JDBCSchemas.scala (hunk 1 above)
Comment (hayssams) [reply src/main/scala/ai/starlake/extract/JDBCSchemas.scala]: You're right. 
Change made

File: src/main/scala/ai/starlake/extract/JdbcDbUtils.scala (hunk 2 above)
Comment (hayssams) [reply src/main/scala/ai/starlake/extract/JdbcDbUtils.scala]: Done

**Comment by tiboun** [comment]
The internal boundaries does apply filter in order to compute them. The output is the same but may implies more executed query

**Pull request merged**